File: lib/domain/entities/task.dart
enum TaskStatus { pending, inProgress, completed }

enum Priority { low, medium, high, urgent }

class Task {
  final String id;
  final String projectId;
  final String? parentTaskId;
  final String title;
  final String description;
  final Priority priority;
  final TaskStatus status;
  final DateTime? startDate;
  final DateTime? dueDate;
  final List<String> tags;
  final int? estimatedMinutes;
  final int? actualMinutes;
  final bool isRecurring;
  final String? recurringRule;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Task({
    required this.id,
    required this.projectId,
    this.parentTaskId,
    required this.title,
    this.description = '',
    this.priority = Priority.medium,
    this.status = TaskStatus.pending,
    this.startDate,
    this.dueDate,
    this.tags = const [],
    this.estimatedMinutes,
    this.actualMinutes,
    this.isRecurring = false,
    this.recurringRule,
    required this.createdAt,
    required this.updatedAt,
  });

  Task copyWith({
    String? id,
    String? projectId,
    String? parentTaskId,
    String? title,
    String? description,
    Priority? priority,
    TaskStatus? status,
    DateTime? startDate,
    DateTime? dueDate,
    List<String>? tags,
    int? estimatedMinutes,
    int? actualMinutes,
    bool? isRecurring,
    String? recurringRule,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Task(
      id: id ?? this.id,
      projectId: projectId ?? this.projectId,
      parentTaskId: parentTaskId ?? this.parentTaskId,
      title: title ?? this.title,
      description: description ?? this.description,
      priority: priority ?? this.priority,
      status: status ?? this.status,
      startDate: startDate ?? this.startDate,
      dueDate: dueDate ?? this.dueDate,
      tags: tags ?? this.tags,
      estimatedMinutes: estimatedMinutes ?? this.estimatedMinutes,
      actualMinutes: actualMinutes ?? this.actualMinutes,
      isRecurring: isRecurring ?? this.isRecurring,
      recurringRule: recurringRule ?? this.recurringRule,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/src/models/task.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use chrono::{DateTime, Utc, FixedOffset};

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Task {
    pub id: String,
    pub user_id: String,
    pub project_id: String,
    pub parent_task_id: Option<String>,
    pub title: String,
    pub description: String,
    pub priority: i32,
    pub status: i32,
    pub start_date: Option<DateTime<FixedOffset>>,
    pub due_date: Option<DateTime<FixedOffset>>,
    pub tags: String,
    pub estimated_minutes: Option<i32>,
    pub actual_minutes: Option<i32>,
    pub is_recurring: bool,
    pub recurring_rule: Option<String>,
    pub created_at: DateTime<FixedOffset>,
    pub updated_at: DateTime<FixedOffset>,
    pub sort_order: i32,
    pub deleted_at: Option<DateTime<FixedOffset>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateTask {
    pub project_id: String,
    pub parent_task_id: Option<String>,
    pub title: String,
    #[serde(default = "default_desc")]
    pub description: String,
    #[serde(default = "default_priority")]
    pub priority: i32,
    #[serde(default)]
    pub status: i32,
    pub start_date: Option<DateTime<FixedOffset>>,
    pub due_date: Option<DateTime<FixedOffset>>,
    #[serde(default = "default_tags")]
    pub tags: String,
    pub estimated_minutes: Option<i32>,
    pub actual_minutes: Option<i32>,
    #[serde(default)]
    pub is_recurring: bool,
    pub recurring_rule: Option<String>,
    #[serde(default)]
    pub sort_order: i32,
}

fn default_desc() -> String {
    "".to_string()
}

fn default_priority() -> i32 {
    2
}

fn default_tags() -> String {
    "[]".to_string()
}